pandas/core/reshape/merge.py

```diff
@@ -2407,13 +2407,16 @@ def _factorize_keys(
                 or is_string_dtype(lk.dtype)
                 and not sort
             )
+            or (is_string_dtype(lk.dtype) and lk.dtype.storage == "pyarrow")  # type: ignore[attr-defined]  # noqa: E501
```

Comment:
would using `isinstance(lk.dtype, StringDtype)` fix the mypy complaint?  (and possibly be more robust to object dtype cases?)

Reply:
You still have to check the storage (python strings...)

Reply:
yes but doing the isinstace check instead of is_string_dtype then makes the mypy complaint go away?  (and excludes object dtype and is just better in general)

Reply:
ah got you, done